frontend/src/scenes/insights/insightSceneLogic.tsx

```diff
@@ -132,6 +133,15 @@ export const insightSceneLogic = kea<insightSceneLogicType>({
                 actions.setSceneState(insightId, insightMode)
                 if (insightId !== oldInsightId && insightId === 'new') {
                     actions.createNewInsight(filters)
+                    return
+                }
+                // Redirect #filters={} to just /edit.
+                if (filters && Object.keys(filters).length > 0) {
+                    values.insightCache?.logic.actions.setFilters(filters)
+                    router.actions.replace(
+                        insightMode === ItemMode.View ? urls.insightView(insightId) : urls.insightEdit(insightId)
```

Comment:
This is a bit of a an odd experience if the user is on the insight view page. Mainly because there's no "save" button on that page, but we're asking them to click save.

Maybe we always redirect to the edit experience in this case? Or depending on what @clarkus is thinking, we could have a slightly different experience for the "view" page

Reply:
Good point! Changed to always redirect to the edit page in this case.